apps/vaporgui/VizWin.cpp

```diff
@@ -858,3 +860,58 @@ void VizWin::updateManip(bool initialize)
 
     GL_ERR_BREAK();
 }
+
+void VizWin::_updateSliceOriginGlyph()
+{
+    VAPoR::SliceParams *sp = dynamic_cast<VAPoR::SliceParams *>(_getRenderParams());
```

Comment:
Hmm.  I have a few questions regarding this file.  Just want to be clear on how to proceed.

1) Querying class types is the same approach we use elsewhere in the code.  For example, we do it when deciding whether to draw a Box/Manip for the Flow seeding tab.  One example is at [line 808](https://github.com/NCAR/VAPOR/blob/15e4643d8f4680af4bb722c81d0d6a839954798a/apps/vaporgui/VizWin.cpp#L808).

`if (gp->ActiveTab() == FlowEventRouter::SeedingTabName) _manipFlowSeedFlag = true;`

Do we want to introduce inconsistency here?  We've talked about refactoring the VizWin and this seems to be adding to that case.

2) An "orientable renderparams flag" does not sound like a parater "tag" that can be changed by the user.  Slice and contour renderers will always be orientable, and this will never change.  Just want to be clear that we're proposing private boolean flag with a getter, and not a tag.

Reply:
> Hmm. I have a few questions regarding this file. Just want to be clear on how to proceed.
> 
> 1. Querying class types is the same approach we use elsewhere in the code.  For example, we do it when deciding whether to draw a Box/Manip for the Flow seeding tab.  One example is at [line 808](https://github.com/NCAR/VAPOR/blob/15e4643d8f4680af4bb722c81d0d6a839954798a/apps/vaporgui/VizWin.cpp#L808).
> 
> `if (gp->ActiveTab() == FlowEventRouter::SeedingTabName) _manipFlowSeedFlag = true;`
> 
> Do we want to introduce inconsistency here? We've talked about refactoring the VizWin and this seems to be adding to that case.
> 

That's a hack and shouldn't have passed review IMHO :-)

> 1. An "orientable renderparams flag" does not sound like a parater "tag" that can be changed by the user.  Slice and contour renderers will always be orientable, and this will never change.  Just want to be clear that we're proposing private boolean flag with a getter, and not a tag.

Not all parameters are settable - they are for information only and do not have setters

Reply:
Hard coding is rarely a good practice, which is what testing the class type amounts to. The contour and slice renderer do not share a base class, so this isn't an option without refactoring. Refactoring simply because contour and and slice both support arbitrary cutting planes would paint us into a corner. What happens is another renderer comes along that requires an orientable plane, but also shares some property with the Barb renderer. Do we then need multiple inheritance?  

Something like a RenderParams::GetOrientable() that defaults to false (so other renderers can ignore it) seems like the way to go here.

Reply:
Got it, makes sense.  Fixed.

Note that the tags XOrigin, YOrigin, and ZOrigin have been moved from SliceParams to RenderParams.  I anticipate we will need to do the same thing with the XRotation, YRotation, and ZRotation tags but I'm going to hold off on this for now until I have a better understanding on how the ContourParams will inherit them.